### DISCUSSION TITLE
cannot find function "FeaturePlot"

### DISCUSSION THREAD

**Discussion by Joriene382** [satijalab/seurat#7689] [Q&A] [answered]
Hi,

We keep running into this same issue. When trying to run the FeaturePlot function in r (using windows on multiple PCs) we almost always get the same error message:

> FeaturePlot(PSN.proprio, features = c("nCount_RNA"))
Error in FeaturePlot(PSN.proprio, features = c("nCount_RNA")) : 
  could not find function "FeaturePlot"

We know this function is installed (and we got it to work twice using the same code strings). Does anyone have any suggestions how to solve or trouble shoot this? Could this be some sort of incompatibility with a windows program that may be running in the background? 

Thanks,
Joriene

**Comment by samuel-marsh** [comment]
Hi Joriene,

This would indicate that you have not loaded the Seurat package in your R session.  What is the output of `sessionInfo()` when you get this error?

Best,
Sam

**Comment by Joriene382** [comment]
Hi Sam,

Thanks for the reply. We have loaded Seurat and when we check this in r studio using:

system.file(package='Seurat')
library(Seurat)
lsf.str("package:Seurat")

we also find that the FeaturePlot function is there. (Also we have been able to get this to work but only about 2 out of a ~50 times of trying, all with the same code).

Thanks
Joriene

From: Samuel Marsh ***@***.***>
Sent: Wednesday, August 16, 2023 4:47 PM
To: satijalab/seurat ***@***.***>
Cc: De Nooij, Joriene C. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [satijalab/seurat] cannot find function "FeaturePlot" (Discussion #7689)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Hi Joriene,

This would indicate that you have not loaded the Seurat package in your R session. What is the output of sessionInfo() when you get this error?

Best,
Sam

-

**Reply by samuel-marsh** [reply]
Can you post output of `sessionInfo()`?

**Comment by Joriene382** [comment]
Sorry - not that fluent in r yet.
See several options but can you specify what code to use for this?
Thks, J


From: Samuel Marsh ***@***.***>
Sent: Thursday, August 17, 2023 6:36 AM
To: satijalab/seurat ***@***.***>
Cc: De Nooij, Joriene C. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [satijalab/seurat] cannot find function "FeaturePlot" (Discussion #7689)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Can you post output of sessionInfo()?

-

**Reply by samuel-marsh** [reply]
Hi,

No problem, so after you get error about FeaturePlot not found then just copy and paste code here: `sessionInfo()`

and then paste the output here.

Best,
Sam

**Comment by Joriene382** [comment]
Thanks - plse see below:

> FeaturePlot(PSN.proprio, features = c("nCount_RNA"))
Error in FeaturePlot(PSN.proprio, features = c("nCount_RNA")) :
  could not find function "FeaturePlot"

> sessionInfo()
R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
  [1] deldir_1.0-9           pbapply_1.7-2          gridExtra_2.3          rlang_1.1.1            magrittr_2.0.3
  [6] RcppAnnoy_0.0.21       spatstat.geom_3.2-4    matrixStats_1.0.0      ggridges_0.5.4         compiler_4.3.0
 [11] png_0.1-8              vctrs_0.6.3            reshape2_1.4.4         stringr_1.5.0          pkgconfig_2.0.3
 [16] fastmap_1.1.1          ellipsis_0.3.2         utf8_1.2.3             promises_1.2.1         purrr_1.0.2
 [21] jsonlite_1.8.7         goftest_1.2-3          later_1.3.1            spatstat.utils_3.0-3   irlba_2.3.5.1
 [26] parallel_4.3.0         cluster_2.1.4          R6_2.5.1               ica_1.0-3              stringi_1.7.12
 [31] RColorBrewer_1.1-3     spatstat.data_3.0-1    reticulate_1.31        parallelly_1.36.0      lmtest_0.9-40
 [36] scattermore_1.2        Rcpp_1.0.11            tensor_1.5             future.apply_1.11.0    zoo_1.8-12
 [41] sctransform_0.3.5      httpuv_1.6.11          Matrix_1.6-0           splines_4.3.0          igraph_1.5.1
 [46] tidyselect_1.2.0       abind_1.4-5            spatstat.random_3.1-5  codetools_0.2-19       miniUI_0.1.1.1
 [51] spatstat.explore_3.2-1 listenv_0.9.0          lattice_0.21-8         tibble_3.2.1           plyr_1.8.8
 [56] shiny_1.7.4.1          ROCR_1.0-11            Rtsne_0.16             future_1.33.0          survival_3.5-5
 [61] polyclip_1.10-4        fitdistrplus_1.1-11    pillar_1.9.0           BiocManager_1.30.22    Seurat_4.3.0.1
 [66] KernSmooth_2.23-22     plotly_4.10.2          generics_0.1.3         sp_2.0-0               ggplot2_3.4.2
 [71] munsell_0.5.0          scales_1.2.1           globals_0.16.2         xtable_1.8-4           glue_1.6.2
 [76] lazyeval_0.2.2         tools_4.3.0            data.table_1.14.8      RANN_2.6.1             leiden_0.4.3
 [81] cowplot_1.1.1          grid_4.3.0             tidyr_1.3.0            colorspace_2.1-0       nlme_3.1-163
 [86] patchwork_1.1.2        cli_3.6.1              spatstat.sparse_3.0-2  fansi_1.0.4            viridisLite_0.4.2
 [91] dplyr_1.1.2            uwot_0.1.16            gtable_0.3.3           digest_0.6.33          progressr_0.14.0
 [96] ggrepel_0.9.3          htmlwidgets_1.6.2      SeuratObject_4.1.3     htmltools_0.5.6        lifecycle_1.0.3
[101] httr_1.4.6             mime_0.12              MASS_7.3-60
>

From: Samuel Marsh ***@***.***>
Sent: Thursday, August 17, 2023 8:15 AM
To: satijalab/seurat ***@***.***>
Cc: De Nooij, Joriene C. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [satijalab/seurat] cannot find function "FeaturePlot" (Discussion #7689)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Hi,

No problem, so after you get error about FeaturePlot not found then just copy and paste code here: sessionInfo()

and then paste the output here.

Best,
Sam

-

**Reply by samuel-marsh** [reply, marked as answer]
Hi,

Thanks!  So what this is showing is that the Seurat package has not actually been loaded into the environment, which is why R cannot find the function `FeaturePlot`.  Try restarting your R session and then before running any other code run: `library(Seurat)` and that should solve the issue.

Best,
Sam

**Comment by Joriene382** [comment]
It worked! Thanks so much!
But just for understanding, we certainly loaded all these packages and libraries (as the four below) when starting a new session and then still got the 'function not found' error. Do you mean that we have to run the "library(Seurat)" line every time we start a session but especially just before we run the feature plot function? (sorry for being so ignorant)

install.packages("dplyr")
library(dplyr)

install.packages("Seurat")
library(Seurat)

install.packages("patchwork")
library(patchwork)

install.packages("Signac")
library(Signac)

From: Samuel Marsh ***@***.***>
Sent: Thursday, August 17, 2023 8:41 AM
To: satijalab/seurat ***@***.***>
Cc: De Nooij, Joriene C. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [satijalab/seurat] cannot find function "FeaturePlot" (Discussion #7689)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Hi,

Thanks! So what this is showing is that the Seurat package has not actually been loaded into the environment, which is why R cannot find the function FeaturePlot. Try restarting your R session and then before running any other code run: library(Seurat) and that should solve the issue.

Best,
Sam

-

**Comment by samuel-marsh** [comment]
No problem!  Every time you start new R session you need to run `library(X)` to load the packages you need.  Then don't need to run that again until starting new session.

Best,
Sam

**Comment by Joriene382** [comment]
Don't mean to waste your time further but are there code lines that can 'undo' the library(x)? So if you run these after you will have to rerun library (x)?

(Just asking because we really had many session where we reran all the code (installing packages, running libraries) before trying to run the featureplot.

From: Samuel Marsh ***@***.***>
Sent: Thursday, August 17, 2023 8:56 AM
To: satijalab/seurat ***@***.***>
Cc: De Nooij, Joriene C. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [satijalab/seurat] cannot find function "FeaturePlot" (Discussion #7689)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

No problem! Every time you start new R session you need to run library(X) to load the packages you need. Then don't need to run that again until starting new session.

Best,
Sam

-

**Reply by samuel-marsh** [reply]
Hi,

So you don't need to install packages every time unless you are on new machine or new version of R.

The only ways to unload packages are using things like `detach` and `unloadNamespace` so it shouldn't be occurring unless you do that

**Comment by Joriene382** [comment]
Ok - Thanks again for all your help!

Joriene

From: Samuel Marsh ***@***.***>
Sent: Thursday, August 17, 2023 9:11 AM
To: satijalab/seurat ***@***.***>
Cc: De Nooij, Joriene C. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [satijalab/seurat] cannot find function "FeaturePlot" (Discussion #7689)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Hi,

So you don't need to install packages every time unless you are on new machine or new version of R.

The only ways to unload packages are using things like detach and unloadNamespace so it shouldn't be occurring unless you do that

-

**Reply by samuel-marsh** [reply]
Happy to help!
